packages/broker/test/integration/plugins/operator/VoteOnSusppectNodeService.test.ts

```diff
@@ -0,0 +1,131 @@
+        await (await sponsorship.connect(flagger.operatorWallet).sponsor(parseEther('500'))).wait()
+
+        logger.debug('each operator delegates to its operactor contract')
+        logger.debug('delegating from flagger: ' + flagger.operatorWallet.address)
+        await (await token.connect(flagger.operatorWallet).transferAndCall(flagger.operatorContract.address,
+            parseEther('200'), flagger.operatorWallet.address)).wait()
+        logger.debug('delegating from target: ' + target.operatorWallet.address)
+        await (await token.connect(target.operatorWallet).transferAndCall(target.operatorContract.address,
+            parseEther('200'), target.operatorWallet.address)).wait()
+        logger.debug('delegating from voter: ' + voter.operatorWallet.address)
+        await (await token.connect(voter.operatorWallet).transferAndCall(voter.operatorContract.address,
+            parseEther('200'), voter.operatorWallet.address)).wait()
+        
+        logger.debug('staking to sponsorship contract from flagger and target and voter')
+        logger.debug('staking from flagger: ' + flagger.operatorContract.address)
+        await (await flagger.operatorContract.stake(sponsorship.address, parseEther('150'))).wait()
+        logger.debug('staking from target: ' + target.operatorContract.address)
+        await (await target.operatorContract.stake(sponsorship.address, parseEther('150'))).wait()
+        logger.debug('staking from voter: ' + voter.operatorContract.address)
+        await (await voter.operatorContract.stake(sponsorship.address, parseEther('150'))).wait()
+
+        logger.debug('registering node addresses')
+        await (await flagger.operatorContract.setNodeAddresses([await flagger.operatorContract.owner()])).wait()
+
+        logger.debug('flagging target operator')
+        const mockVoteOnSuspectNodeHelper = mock<VoteOnSuspectNodeHelper>()
+        mockVoteOnSuspectNodeHelper.voteOnFlag.mockImplementation(async (operatorAddress, suspectAddress, flag) => {
+            logger.debug('mockVoteOnSuspectNodeHelper.voteOnFlag called')
+            logger.debug('operatorAddress: ' + operatorAddress)
+            logger.debug('suspectAddress: ' + suspectAddress)
+            logger.debug('flag: ' + flag)
+        })
+        // @ts-expect-error mock
+        voterVoteService.voteOnSuspectNodeHelper = mockVoteOnSuspectNodeHelper
+        await (await flagger.operatorContract.flag(sponsorship.address, target.operatorContract.address)).wait()
+        // check that voter votes
+        await waitForCondition(async () => {
+            return mockVoteOnSuspectNodeHelper.voteOnFlag.mock.calls.length > 0 &&
+            mockVoteOnSuspectNodeHelper.voteOnFlag.mock.calls[0][0] === sponsorship.address &&
+            mockVoteOnSuspectNodeHelper.voteOnFlag.mock.calls[0][1] === target.operatorContract.address &&
+            mockVoteOnSuspectNodeHelper.voteOnFlag.mock.calls[0][2] === true
+        }, 10000, 1000)
```

Comment:
I'd simplify this down to
```suggestion
        // @ts-expect-error mock
        voterVoteService.voteOnSuspectNodeHelper = mockVoteOnSuspectNodeHelper
        await (await flagger.operatorContract.flag(sponsorship.address, target.operatorContract.address)).wait()
        // check that voter votes
        await waitForCondition(() => mockVoteOnSuspectNodeHelper.voteOnFlag.mock.calls.length > 0 , 10000)
        expect(mockVoteOnSuspectNodeHelper.voteOnFlag).toHaveBeenCalledTimes(1)
        expect(mockVoteOnSuspectNodeHelper.voteOnFlag).toHaveBeenCalledWith(sponsorship.address, target.operatorContract.address, true)
```